### ISSUE TITLE
Move UX server side code out of APM UI

### ISSUE THREAD

**Issue by shahzad31** [elastic/uptime#454]
We still have node server side code in the APM UI for user experience app, we need to move that out of APM UI.

This is a follow up to https://github.com/elastic/kibana/issues/78430

There are two approaches that will be taken to accomplish this. 

User exploratory view to replace charts, this will auto remove server side routes and will make charts completely client side.

In cases where we can't use exploratory view, we should use async search api to make calls to Elasticsearch.

EDIT by @justinkambic:

## Areas of focus

- [ ] ~~Create common filters shared state~~
- [x] Service list (web application) - use data plugin search https://github.com/elastic/kibana/pull/132548
- [x] ~Percentile - data plugin~ There's no reference to APM routes for the percentile component
- [x] Environment vars - data plugin https://github.com/elastic/kibana/pull/134393
- [x] Filter search input - data plugin https://github.com/elastic/kibana/pull/132654
- [x] Client metrics - data plugin - https://github.com/elastic/kibana/pull/133322
- [x] Metrics- data plugin https://github.com/elastic/kibana/pull/133974
- [x] Core web vitals - data plugin https://github.com/elastic/kibana/pull/133974
- [x] Page load and total page views - exploratory view embeddable https://github.com/elastic/kibana/pull/132970
  - [x] Page views: https://github.com/elastic/kibana/pull/134758
  - [x] Page load distribution- data plugin for now, [until Lens unblocks it](https://github.com/elastic/kibana/issues/133991): https://github.com/elastic/kibana/pull/135072
- [x] Visitor breakdowns - lens embeddable - https://github.com/elastic/kibana/pull/134684
- [x] Errors table - data plugin - https://github.com/elastic/kibana/pull/133040
- [x] Audit APM for any leftover UX route code
  - [x] `hasRumData` route: https://github.com/elastic/kibana/pull/134524
  - [x] `long-task-metrics` [route](https://github.com/elastic/kibana/blob/main/x-pack/plugins/ux/public/components/app/rum_dashboard/ux_metrics/key_ux_metrics.tsx#L58):  - https://github.com/elastic/kibana/pull/134711#issuecomment-1159760749
---

Additional items:
- [x] Explore removal of projections from UX components (APM no longer uses them): covered as part of https://github.com/elastic/kibana/pull/135072, which includes the final route removal
  - [x] Move `processor_event` to shared o11y plugin and replace individual implementations, per [this comment](https://github.com/elastic/kibana/pull/132548#discussion_r877817456) https://github.com/elastic/kibana/pull/135367
- [x] Add e2e tests for page view chart https://github.com/elastic/kibana/pull/135352

**Comment by paulb-elastic** [comment]
Pushing to 8.3, and the expectation is that the simplest implementation is done for this to get out of the APM UI codebase

**Comment by paulb-elastic** [comment]
Moving to 8.4 as these PRs are not being merged on for 8.3

**Comment by emilioalvap** [comment]
@shahzad31 @justinkambic Checking percentile component, it seems to be [statically declared](https://github.com/elastic/kibana/blob/main/x-pack/plugins/ux/public/components/app/rum_dashboard/user_percentile/index.tsx) inside the UX plugin. Can we remove that from the task list or is there anything else that needs migrating here?

**Comment by paulb-elastic** [comment]
Mostly done but moving over to 8.5 board where the last remaining item https://github.com/elastic/kibana/pull/135367 will be targeted for.

**Comment by emilioalvap** [comment]
Marking as done, last PR was merged into 8.5

**Comment by dominiqueclarke** [comment]
POST FF Testing:

Did general smoke tests of the UX page on 8.5 local branch via data from oblt-cli.

Looks great, generally, but the browser breakdown and operating system breakdown appears to be broken.

<img width="753" alt="Screen Shot 2022-09-28 at 3 46 16 PM" src="https://user-images.githubusercontent.com/11356435/192875003-446d41ac-7c8c-4918-9439-4abbb7715546.png">
cc: @justinkambic @emilioalvap

**Comment by paulb-elastic** [comment]
Reopening as per @dominiqueclarke's discovery

**Comment by emilioalvap** [comment]
Fixed by elastic/kibana#142992.

Moving to Done Done, as @shahzad31 and I have validated it.
